app/models/website_resolver.rb

```diff
@@ -0,0 +1,58 @@
+class WebsiteResolver
+  attr_reader :results
+
+  def self.resolve(url)
+    resolver = new(url)
+    resolver.resolve
+    resolver
+  end
+
+  def initialize(url)
+    @url = url
+    @results = []
+  end
+
+  def resolve
+    resolve_url if @url
+  end
+
+  def resolved_url
+    @results.last&.fetch :content
+  end
+
+  private
+
+  def resolve_url
+    add_protocol unless protocol?
+    response = connection.get(@url)
+    add_result(response.env)
```

Comment:
Cool, so this is why [`@results.last`](https://github.com/artsy/bearden/pull/184/commits/2edbb0cfa18739683166bcbb06dcfa0784de53e2#diff-39cf08e9f94dcacae44558329579bafdR20) returns the 200 or 404.